poetry/utils/exporter.py

```diff
@@ -94,14 +94,16 @@ def _export_requirements_txt(
                 or dependency.is_directory()
             )
 
-            if is_direct_reference:
-                line = requirement
-            else:
-                line = "{}=={}".format(package.name, package.version)
+            if not is_direct_reference:
+                line += "{}=={}".format(package.name, package.version)
                 if ";" in requirement:
                     markers = requirement.split(";", 1)[1].strip()
                     if markers:
                         line += "; {}".format(markers)
+            elif dependency.is_directory():  # pip doesn't support ` @ ` for directories
```

Comment:
I do not think this should be done, because directory dependencies can be PEP 517 dependencies, which means `-e` will fail.

Reply:
I don't follow. Having a line `foo @ some/path` in `requirements.txt` will break `pip install` with `ERROR: Invalid requirement: 'foo @ some/path'`? Should the exporter add a new PEP508 format besides `requirements.txt`?

Reply:
I think what is being discussed in #3121 is a more complete solution. But yes, the comments in #3118 are more current.

Should this be closed in favour of that?